### PR TITLE
Create upload images functionality

### LINKED ISSUE #128: [React.js ] upload images feature

### PR DESCRIPTION
**Pull request by tariqabusada**
- Startup storage on firebase and configurations for it.
- Create component to use it to upload images.

### CODE REVIEW

File: client/src/components/UpLoadImages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    setFile(list);
+  };
+  const handleUpload = () => {
+    if (!file) {
+      setSnackbar(true);
+    } else {
+      setLoading(true);
+      const toUploadfiles = [];
+      for (let i = 0; i < file.length; i += 1) {
+        const storageRef = ref(storage, `/Images/cars/carId/${file[i].name}`);
```
Comment (LinaYahya) [inline client/src/components/UpLoadImages/index.tsx:41]:
```suggestion
        const storageRef = ref(storage, `/Images/cars/${carId}/${file[i].name}`);
```

File: client/src/components/UpLoadImages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        const touploadFile = uploadBytes(storageRef, file[i]);
+        toUploadfiles.push(touploadFile);
+      }
+      Promise.all(toUploadfiles)
+        .then((d) => {
```
Comment (LinaYahya) [inline client/src/components/UpLoadImages/index.tsx]:
```suggestion
        .then((snapshots) => {
```

File: client/src/components/UpLoadImages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          const getImages = d.map(({ ref: snapshotRef }) => getDownloadURL(snapshotRef));
+          return getImages;
+        })
+        .then((t) => Promise.all(t))
```
Comment (LinaYahya) [inline client/src/components/UpLoadImages/index.tsx]:
```suggestion
        .then((idownloadImgsPromise) => Promise.all(t))
```

File: client/src/components/UpLoadImages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      <input ref={fileInput} multiple type="file" onChange={handleChange} accept="/image/*" />
+      {loading
+        ? (
+          <>
+            <Button disabled>Upload</Button>
+            <p>Uploading ...</p>
+          </>
+        )
+        : <Button onClick={handleUpload}>Upload</Button>}
+      <Snackbar open={snackbar} autoHideDuration={4000} onClose={handleCloseSnack}>
```
Comment (LinaYahya) [inline client/src/components/UpLoadImages/index.tsx]: why to use three `snackbars`, simplly we can use one component we create check `customSnackbar` component

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by mai-mohammed — APPROVED** [review]

**Review by husamkamal — APPROVED** [review]

**Review by abdou059 — APPROVED** [review]

**Pull request merged**